x-pack/plugins/security_solution/public/management/pages/host_isolation_exceptions/view/components/form_flyout.tsx

```diff
@@ -21,241 +21,181 @@ import {
-  const toasts = useToasts();
-  const location = useHostIsolationExceptionsSelector(getCurrentLocation);
-  const creationInProgress = useHostIsolationExceptionsSelector((state) =>
-    isLoadingResourceState(state.form.status)
-  );
-  const creationSuccessful = useHostIsolationExceptionsSelector((state) =>
-    isLoadedResourceState(state.form.status)
-  );
-  const creationFailure = useHostIsolationExceptionsSelector(getFormStatusFailure);
-  const exceptionToEdit = useHostIsolationExceptionsSelector(getExceptionToEdit);
-  const navigateCallback = useHostIsolationExceptionsNavigateCallback();
-  const queryClient = useQueryClient();
-  const history = useHistory();
-
-  const [formHasError, setFormHasError] = useState(true);
-  const [exception, setException] = useState<
-    CreateExceptionListItemSchema | UpdateExceptionListItemSchema | undefined
-  >(undefined);
-
-  const onCancel = useCallback(
-    () =>
-      navigateCallback({
-        show: undefined,
-        id: undefined,
-      }),
-    [navigateCallback]
-  );
+export const HostIsolationExceptionsFormFlyout = memo(
+  ({ onCancel, id }: { onCancel: () => void; id?: string }) => {
+    const http = useHttp();
+    const toasts = useToasts();
+    const queryClient = useQueryClient();
 
-  // load data to edit or create
-  useEffect(() => {
-    if (location.show === 'create' && exception === undefined) {
-      setException(createEmptyHostIsolationException());
-    } else if (location.show === 'edit') {
-      // prevent flyout to show edit without an id
-      if (!location.id) {
-        onCancel();
-        return;
-      }
-      // load the exception to edit
-      if (!exceptionToEdit || location.id !== exceptionToEdit.id) {
-        dispatch({
-          type: 'hostIsolationExceptionsMarkToEdit',
-          payload: { id: location.id },
-        });
-      } else if (exception === undefined) {
-        setException(exceptionToEdit);
-      }
-    }
-  }, [dispatch, exception, exceptionToEdit, location.id, location.show, onCancel]);
+    const isEditing = id !== undefined;
+    const [exception, setException] = useState<
+      CreateExceptionListItemSchema | UpdateExceptionListItemSchema | undefined
+    >(undefined);
 
-  // handle creation and edit success
-  useEffect(() => {
-    if (creationSuccessful && exception?.name) {
-      onCancel();
-      dispatch({
-        type: 'hostIsolationExceptionsFormStateChanged',
-        payload: {
-          type: 'UninitialisedResourceState',
+    useQuery<UpdateExceptionListItemSchema | CreateExceptionListItemSchema, ServerApiError>(
+      ['hostIsolationExceptions', 'form', id],
+      async () => {
+        // for editing, fetch from the API
+        if (id !== undefined) {
+          return getOneHostIsolationExceptionItem(http, id);
+        }
+        // for adding, return a new empty object
+        return createEmptyHostIsolationException();
```

Comment:
nit: Could be this the default value instead?

Reply:
I thought about it but then when you edit an item you will not see the loading bar instead you'll be presented with an emtpy form and then a second or so later it will show content.

When you put initial data into useQuery it will never show the `loading` state.